### PR TITLE
🐛 [Bug] :  long comments can be folded up and down

### LINKED ISSUE #844: 🐛 [Bug] Comments cannot be expanded if they are too long
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-06-17T11:36:30Z" title="Monday, June 17th 2019, 1:36:30 pm +02:00">Jun 17, 2019</time>_
_Closed <time datetime="2019-07-03T08:52:20Z" title="Wednesday, July 3rd 2019, 10:52:20 am +02:00">Jul 3, 2019</time>_
---

### Comments cannot be expanded if they are too long

image

Comments are cut off with `...` but cannot be expanded to read the full comment.

### Steps to reproduce the behavior
1. Visit Post with Comments
2. See dots on long comments
3. Unable to expand
4. ...
5. Profit

### Expected behavior

Comments should be expandable to read them in full or be displayed in full to begin with.

### Version & Environment
 Type: [Desktop]      
 - OS: [?]      
 - Browser: [Chrome] 

### Additional context

Example: https://nitro-production.human-connection.org/post/5d0107b1cbac47002b176166/prof-rainer-mausfeld-gibt-es-hoffnung-fur-die-lammer

### PR DESCRIPTION
**Pull request by Tirokk**
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-06-19T15:49:11Z" title="Wednesday, June 19th 2019, 5:49:11 pm +02:00">Jun 19, 2019</time>_
_Merged <time datetime="2019-07-03T08:52:20Z" title="Wednesday, July 3rd 2019, 10:52:20 am +02:00">Jul 3, 2019</time>_
---

## 🍰 Pullrequest
long comments can be folded up and down

ezgif com-gif-maker

### Issues
 
- fixes #844 
 
- None

### Todo
- [X] None

### CODE REVIEW

File: webapp/components/Comment.vue [hunk 1]
```diff
@@ -28,13 +27,27 @@
       </no-ssr>
       <!-- eslint-disable vue/no-v-html -->
       <!-- TODO: replace editor content with tiptap render view -->
+
+      <div v-if="isCollapsed" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+      <div
+        v-show="comment.content !== comment.contentExcerpt"
+        style="text-align: right;  margin-right: 20px; margin-top: -12px;"
```
Comment (Tirokk) [inline webapp/components/Comment.vue:34]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-24T19:40:56Z" title="Monday, June 24th 2019, 9:40:56 pm +02:00">Jun 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -23,10 +23,27 @@
     <!-- eslint-disable vue/no-v-html -->
     <!-- TODO: replace editor content with tiptap render view -->
     <ds-space margin-bottom="small" />
-    <div style="padding-left: 40px;" v-html="comment.contentExcerpt" />
-    <!-- eslint-enable vue/no-v-html -->
+    <div v-if="!isActive" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+    <div v-show="comment.content !== comment.contentExcerpt">
+      <span
+        v-if="!isActive"
+        style="padding-left: 40px; color: #17b53f; cursor:pointer"
+        @click="isActive = !isActive"
+      >
+        {{ $t('comment.show.more') }}
+      </span>
+    </div>
+    <div v-if="isActive" v-html="comment.content" style="padding-left: 40px;" />
```


```suggestion
    <div v-else v-html="comment.content" style="padding-left: 40px;" />
```

File: webapp/components/Comment.vue [hunk 2]
```diff
@@ -11,7 +11,6 @@
   </div>
   <div v-else :class="{ comment: true, 'disabled-content': comment.deleted || comment.disabled }">
     <ds-card>
-      <ds-space margin-bottom="small" />
       <ds-space margin-bottom="small">
         <hc-user :user="author" :date-time="comment.createdAt" />
```
Comment (Tirokk) [inline webapp/components/Comment.vue:15]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-24T19:43:55Z" title="Monday, June 24th 2019, 9:43:55 pm +02:00">Jun 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -23,10 +23,27 @@
     <!-- eslint-disable vue/no-v-html -->
     <!-- TODO: replace editor content with tiptap render view -->
     <ds-space margin-bottom="small" />
-    <div style="padding-left: 40px;" v-html="comment.contentExcerpt" />
-    <!-- eslint-enable vue/no-v-html -->
+    <div v-if="!isActive" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
```


The logic implemented in this template is calling for it's own component. How about `<collapsible>`?

File: webapp/components/Comment.vue [hunk 3]
```diff
@@ -28,13 +27,27 @@
...
+      <div v-if="isCollapsed" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+      <div
+        v-show="comment.content !== comment.contentExcerpt"
+        style="text-align: right;  margin-right: 20px; margin-top: -12px;"
+      >
+        <a v-if="isCollapsed" style="padding-left: 40px;" @click="isCollapsed = !isCollapsed">
+          {{ $t('comment.show.more') }}
+        </a>
+      </div>
+      <div v-if="!isCollapsed" v-html="comment.content" style="padding-left: 40px;" />
```
Comment (Tirokk) [inline webapp/components/Comment.vue:40]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-24T19:46:37Z" title="Monday, June 24th 2019, 9:46:37 pm +02:00">Jun 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -23,10 +23,27 @@
     <!-- eslint-disable vue/no-v-html -->
     <!-- TODO: replace editor content with tiptap render view -->
     <ds-space margin-bottom="small" />
-    <div style="padding-left: 40px;" v-html="comment.contentExcerpt" />
-    <!-- eslint-enable vue/no-v-html -->
+    <div v-if="!isActive" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+    <div v-show="comment.content !== comment.contentExcerpt">
+      <span
+        v-if="!isActive"
+        style="padding-left: 40px; color: #17b53f; cursor:pointer"
+        @click="isActive = !isActive"
+      >
+        {{ $t('comment.show.more') }}
+      </span>
+    </div>
+    <div v-if="isActive" v-html="comment.content" style="padding-left: 40px;" />
+    <span
+      v-if="isActive"
+      @click="isActive = !isActive"
+      style="padding-left: 40px; color: #17b53f; cursor:pointer"
+    >
+      {{ $t('comment.show.less') }}
```


Could you DRY this out? (DRY == don't repeat yourself, here is a [link](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)). I think if you put the `isCollapsed` property to the root node, that should be it.

File: webapp/components/Comment.vue [hunk 4]
```diff
@@ -28,13 +27,27 @@
       </no-ssr>
```
Comment (Tirokk) [inline webapp/components/Comment.vue:27]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-24T19:47:22Z" title="Monday, June 24th 2019, 9:47:22 pm +02:00">Jun 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -23,10 +23,27 @@
     <!-- eslint-disable vue/no-v-html -->
     <!-- TODO: replace editor content with tiptap render view -->
     <ds-space margin-bottom="small" />
-    <div style="padding-left: 40px;" v-html="comment.contentExcerpt" />
-    <!-- eslint-enable vue/no-v-html -->
+    <div v-if="!isActive" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+    <div v-show="comment.content !== comment.contentExcerpt">
+      <span
+        v-if="!isActive"
```


Call the property `isCollapsed`. From the name `active` would expect only one comment to be active at a time.But `isActive` refers only to sth. visible here.

File: webapp/components/Comment.vue [hunk 5]
```diff
@@ -28,13 +27,27 @@
...
       <!-- TODO: replace editor content with tiptap render view -->
+
+      <div v-if="isCollapsed" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+      <div
+        v-show="comment.content !== comment.contentExcerpt"
+        style="text-align: right;  margin-right: 20px; margin-top: -12px;"
+      >
+        <a v-if="isCollapsed" style="padding-left: 40px;" @click="isCollapsed = !isCollapsed">
+          {{ $t('comment.show.more') }}
+        </a>
```
Comment (Tirokk) [inline webapp/components/Comment.vue:38]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-06-28T05:55:49Z" title="Friday, June 28th 2019, 7:55:49 am +02:00">Jun 28, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -23,10 +23,27 @@
     <!-- eslint-disable vue/no-v-html -->
     <!-- TODO: replace editor content with tiptap render view -->
     <ds-space margin-bottom="small" />
-    <div style="padding-left: 40px;" v-html="comment.contentExcerpt" />
-    <!-- eslint-enable vue/no-v-html -->
+    <div v-if="!isActive" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+    <div v-show="comment.content !== comment.contentExcerpt">
+      <span
+        v-if="!isActive"
+        style="padding-left: 40px; color: #17b53f; cursor:pointer"
+        @click="isActive = !isActive"
+      >
+        {{ $t('comment.show.more') }}
+      </span>
+    </div>
+    <div v-if="isActive" v-html="comment.content" style="padding-left: 40px;" />
+    <span
+      v-if="isActive"
+      @click="isActive = !isActive"
+      style="padding-left: 40px; color: #17b53f; cursor:pointer"
```


I would imagine, if there is a way, use the color from the style guide as color name via CSS?
I think I have seen this at other places …

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-02T15:18:33Z" title="Tuesday, July 2nd 2019, 5:18:33 pm +02:00">Jul 2, 2019</time>_

---


should we pair on testing this @ogerly

File: webapp/components/Comment.vue [hunk 6]
```diff
@@ -28,13 +27,27 @@
...
+      </div>
+      <div v-if="!isCollapsed" v-html="comment.content" style="padding-left: 40px;" />
+      <div style="text-align: right;  margin-right: 20px; margin-top: -12px;">
+        <a v-if="!isCollapsed" @click="isCollapsed = !isCollapsed" style="padding-left: 40px; ">
+          {{ $t('comment.show.less') }}
+        </a>
+      </div>
       <ds-space margin-bottom="small" />
-      <div style="padding-left: 40px;" v-html="comment.contentExcerpt" />
-      <!-- eslint-enable vue/no-v-html -->
```
Comment (Tirokk) [inline webapp/components/Comment.vue:33]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-03T05:59:54Z" title="Wednesday, July 3rd 2019, 7:59:54 am +02:00">Jul 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -22,11 +22,33 @@
     </no-ssr>
     <!-- eslint-disable vue/no-v-html -->
     <!-- TODO: replace editor content with tiptap render view -->
-    <ds-space margin-bottom="small" />
-    <div style="padding-left: 40px;" v-html="comment.contentExcerpt" />
-    <!-- eslint-enable vue/no-v-html -->
+
+    <div v-if="isCollapsed" v-html="comment.contentExcerpt" style="padding-left: 40px;" />
+    <div
+      v-show="comment.content !== comment.contentExcerpt"
+      style="text-align: right;  margin-right: 20px; margin-top: -12px;"
+    >
+      <span
+        v-if="isCollapsed"
+        style="padding-left: 40px; color: #17b53f; cursor:pointer"
+        @click="isCollapsed = !isCollapsed"
+      >
+        {{ $t('comment.show.more') }}
+      </span>
+    </div>
+    <div v-if="!isCollapsed" v-html="comment.content" style="padding-left: 40px;" />
+    <div style="text-align: right;  margin-right: 20px; margin-top: -12px;">
+      <span
+        v-if="!isCollapsed"
+        @click="isCollapsed = !isCollapsed"
+        style="padding-left: 40px; color: #17b53f; cursor:pointer"
+      >
+        {{ $t('comment.show.less') }}
+      </span>
+    </div>
```


Like @roschaefer said. Would be nice to make a component out this.
But my opinion is to make an issue for the future, after we replaced the alpha. @ogerly @roschaefer 

PS: And we have to use the editor functionality to display this content in the future, if I have understood @appinteractive right. Therefor this has to be consider if we do make an own component.

File: webapp/components/Comment.vue (hunk 1 above)
Comment (Tirokk) [reply webapp/components/Comment.vue:34]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-03T06:29:31Z" title="Wednesday, July 3rd 2019, 8:29:31 am +02:00">Jul 3, 2019</time>_

---


There is a show in-between. So this easy and quick solution is not working.
One could code this a bit more elegant, but I didn’t like to pressure here, just for principle.

File: webapp/components/Comment.vue (hunk 2 above)
Comment (Tirokk) [reply webapp/components/Comment.vue:15]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-03T06:35:09Z" title="Wednesday, July 3rd 2019, 8:35:09 am +02:00">Jul 3, 2019</time>_

---


We have to refactor this anyway by using the editor library to display the content very soon. We should do this than together …

File: webapp/components/Comment.vue (hunk 3 above)
Comment (Tirokk) [reply webapp/components/Comment.vue:40]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-03T06:36:35Z" title="Wednesday, July 3rd 2019, 8:36:35 am +02:00">Jul 3, 2019</time>_

---


Same same … as said before. Not just this …

File: webapp/components/Comment.vue (hunk 4 above)
Comment (Tirokk) [reply webapp/components/Comment.vue:27]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-03T06:36:52Z" title="Wednesday, July 3rd 2019, 8:36:52 am +02:00">Jul 3, 2019</time>_

---


Done

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-03T06:47:11Z" title="Wednesday, July 3rd 2019, 8:47:11 am +02:00">Jul 3, 2019</time>_

---


Please pull in the master, than I will merge …

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-07-03T08:07:12Z" title="Wednesday, July 3rd 2019, 10:07:12 am +02:00">Jul 3, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/864?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@76388d4`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

**Review by Mogge — CHANGES_REQUESTED** [review]
Hey Alex,

you have done a great work!!! 💫😎
Really important feature you solved.
You know VueJS better and better! 😍

Take just a little step and you got it finished … 🤗

**Review by Mogge — APPROVED** [review]
Hey Alex,
super that you made this feature getting reality! 🚀

We rock it 🤘🏼

**Pull request closed without merging**